arrow-libs/fx/arrow-fx-coroutines/src/jvmTest/kotlin/arrow/fx/coroutines/FlowJvmTest.kt

```diff
@@ -59,4 +67,91 @@ class FlowJvmTest : ArrowFxSpec(spec = {
       }
     }
   }
+
+  "fixedDelay" {
+    runBlockingTest {
+      checkAll(Arb.positiveInts().map(Int::toLong), Arb.int(1..100)) { waitPeriod, n ->
+        val emissionDuration = waitPeriod / 10L
+        var state: Long? = null
+
+        val rate = flow { emit(delay(Duration.milliseconds(waitPeriod))) }.repeat()
+          .map {
+            val now = state ?: currentTime
+            val nextNow = currentTime
+            val lapsed = nextNow - now
+            state = nextNow
+            delay(emissionDuration)
+            lapsed
+          }
+          .take(n)
+          .toList()
+
+        rate.first() shouldBe 0 // First element is immediately
+        rate.drop(1).forEach { act ->
+          act shouldBe (waitPeriod + emissionDuration) // Remaining elements all take delay + emission duration
+        }
+      }
+    }
+  }
+
+  "fixedRate" {
+    runBlockingTest {
+      checkAll(Arb.positiveInts().map(Int::toLong), Arb.int(1..100)) { waitPeriod, n ->
+        val emissionDuration = waitPeriod / 10
+        var state: Long? = null
+
+        val rate = fixedRate(Duration.milliseconds(waitPeriod)) { currentTime }
+          .map {
+            val now = state ?: currentTime
+            val nextNow = currentTime
+            val lapsed = nextNow - now
+            state = nextNow
+            delay(emissionDuration)
+            lapsed
+          }
+          .take(n)
+          .toList()
+
+        rate.first() shouldBe 0 // First element is immediately
+        rate.drop(1).forEach { act ->
+          // Remaining elements all take total of waitPeriod, emissionDuration is correctly taken into account.
+          act shouldBe waitPeriod
+        }
+      }
+    }
+  }
+
+  "fixedRate(dampen = true)" {
+    val waitPeriod = 1000L
+    val n = 3
+    val timeout = (n + 1) * waitPeriod + 500
```

Comment:
This seems like an extremely brittle test :/ Is this not possible with `runBlockingTest`?

Reply:
Yes, I had the same feeling.. There however currently a bug in `runBlockingTest` preventing us to write this test with it

Since I wanted to have a test for `dampen` I gave this test high delays/timeouts, and hopefully soon it can be turned into a prop-test with `runBlockingTest`.